examples/ex301_solidmechanics.jl

```diff
@@ -0,0 +1,105 @@
+# ------------------------------------------------------------------------------
+# solid mechanics example
+# ------------------------------------------------------------------------------
+
+using LinearAlgebra
+
+# setup
+mesh = Mesh3D(1.0, 1.0, 1.0)
+finep = 3
+coarsep = 2
+dimension = 3
+finebasis = TensorH1LagrangeBasis(finep, finep, dimension)
+coarsebasis = TensorH1LagrangeBasis(coarsep, finep, dimension)
+ctofbasis = TensorH1LagrangeBasis(coarsep, finep, dimension, lagrangequadrature = true)
+
+# constants
+e = 1E6                     # Young's modulus
+ν = 0.3                     # Poisson's ratio
+K = e/(3*(1 - 2*ν))         # bulk modulus
+λ = e*ν/((1 + ν)*(1 - 2*ν)) # Lamé parameters
+μ = e/(2*(1 + ν))
+
+# state
+gradu = zeros(3, 3);
+gradu[1, 1:3] = 1*ones(3);
+gradu[2, 1:3] = 2*ones(3);
+gradu[3, 1:3] = 3*ones(3);
+
+function neohookeanweakform(
+    deltadux::Array{Float64},
+    deltaduy::Array{Float64},
+    deltaduz::Array{Float64},
+    w::Array{Float64},
+)
+    # dP = dF S + F dS
+    #  dS = partialS/partialE
+
+    # deformation gradient
+    F = gradu + I
+    J = det(F)
+    # Green-Lagrange strain tensor
+    E = (gradu*transpose(gradu) + transpose(gradu)*gradu)/2
+    # right Cauchy-Green tensor
+    C = 2*E + I
+    C_inv = C^-1
+    # Second Piola-Kirchhoff
+    S = λ*log(J)*C_inv + μ*(I - C_inv)
+
+    # delta du
+    deltadu = zeros(3, 3)
+    deltadu[1, 1:3] = deltadux
+    deltadu[2, 1:3] = deltaduy
+    deltadu[3, 1:3] = deltaduz
```

Comment:
Is there a reason why to split xyz in the interface and then reassemble here?

Reply:
This is one of those interface rough spots.
I was trying to reduce the confusion over component/derivative that happens in the libCEED interface, and I was trying to provide the same way to set up two different fields or mixed elements.

I'm not sure that I like the current result though.

It looks a little better like this:
```julia
gradu = [
    1*ones(3)';
    2*ones(3)';
    3*ones(3)';
]
//...
    deltadu = [
        deltadux';
        deltaduy';
        deltaduz';
    ]
```

Reply:
A bit more compact: `gradu = [1;2;3] * ones(1,3)`